modules/mauth-authenticator/src/main/java/com/mdsol/mauth/RequestAuthenticator.java

```diff
@@ -168,4 +174,28 @@ private boolean validateSignatureV2 (MAuthRequest mAuthRequest, PublicKey client
     }
   }
 
+  private boolean fallbackValidateSignatureV1(MAuthRequest mAuthRequest, PublicKey clientPublicKey) {
```

Comment:
Is it possible to reuse the logic in `validateSignatureV1`?

Reply:
The method fallbackValidateSignatureV1(MAuthRequest mAuthRequest, PublicKey clientPublicKey) calls validateSignatureV1, but also do pre-post as required:
- pre `validateSignatureV1`:
construct MAuthRequest `mAuthRequestV1` based on the original `mAuthRequest` with V1 headers (getXmwsSignature, getXmwsTime);
- call `validateSignatureV1` with `mAuthRequestV1`;
- post `validateSignatureV1`: write a log message `Completed successful authentication attempt after fallback to V1` if V1 is validated

Reply:
Yes you're right I misread the implementation.